psi4/driver/p4util/exceptions.py

```diff
@@ -374,22 +376,62 @@ class ManagedMethodError(PsiException):
+          - method : str, model/method/level-of-theory
+          - link : str, link address to table in docs
+          - method_type : (str, str), raw value and str repr of governing variable and algorithm
+          - reference : (str, str), raw value and str repr of reference type
+          - qc_module : (str, str), raw value and str repr of targeted implementation
+          - fcae : (str, str), raw value and str repr of all-electron/frozen-core status
 
     """
     message: str
+    stats: Dict[str, Any]
 
     def __init__(self, circs: List[str]):
-        if circs[5] == '':
-            modulemsg = "not available"
-        else:
-            modulemsg = f"not directable to QC_MODULE '{circs[5]}'"
 
-        if len(circs) == 7:
-            msg = f"""{circs[0]}: Method '{circs[1]}' with {circs[2]} '{circs[3]}', FREEZE_CORE '{not circs[6]}', and REFERENCE '{circs[4]}' {modulemsg}"""
+        if circs[0].endswith("_property"):
+            driver = "properties"
+            derivative_int = "prop"  # differs from QCSchema Driver.derivative_int that uses `0` here
+        elif circs[0].endswith("_hessian"):
+            driver = "hessian"
+            derivative_int = 2
+        elif circs[0].endswith("_gradient"):
+            driver = "gradient"
+            derivative_int = 1
         else:
-            msg = f"""{circs[0]}: Method '{circs[1]}' with {circs[2]} '{circs[3]}' and REFERENCE '{circs[4]}' {modulemsg}"""
+            driver = "energy"
+            derivative_int = 0
+
+        docslink = docs_table_link(circs[1], mode=("summary" if circs[5] == "" else "details"))
+        all_electron = circs[6] if (len(circs) == 7) else "(n/a)"
+
+        stats = {
+            # use QCSchema AtomicInput keys where available
+            "driver": driver,
+            "derivative_int": derivative_int,
+            "method": circs[1],
+            "link": docslink,
+            "method_type": (circs[3], f"{circs[2].upper()}={circs[3]}"),
+            "reference": (circs[4], f"REFERENCE={circs[4]}"),
+            "qc_module": (circs[5], f"QC_MODULE={circs[5] or '(auto)'}"),
+            # Note on fcae: most select_* functions don't consider this circumstance. Index 0 of tuple documents this.
+            #   Index 1 of tuple always returns the circumstance, regardless of whether considered, so user can look up availability in table.
+            "fcae": (all_electron, f"FREEZE_CORE={core.get_global_option('FREEZE_CORE')}"),
+        }
+
+        # Note that this message is not generally seen, as driver_util catches and edits it from `stats`
+        conditions2 = [stats[k][1] for k in ["method_type", "reference", "fcae", "qc_module"]]
+        msg = f"Method={stats['method']} is not available for target derivative level (dertype={derivative_int}) under conditions {', '.join(conditions2)}. See {stats['link']}."
+
         PsiException.__init__(self, msg)
-        self.message = '\nPsiException: %s\n\n' % msg
+        self.message = msg
```

Comment:
What's the point of these member variables?

Reply:
Besides `message` (which is primary printing), member variables are a way to send useful info back. So the ConvergenceErrors return the wfn so one could probe how close to convergence it got. The ManagedMethodError is now patterned on RDR's `stats` dict from TDDFT.